engines/python/setup/djl_python/rolling_batch/lmi_dist_v2_rolling_batch.py

```diff
@@ -0,0 +1,104 @@
+
+from collections import OrderedDict
+
+from lmi_dist.api import Request
+from lmi_dist.init_engine import engine_from_args
+from vllm import EngineArgs, SamplingParams
+
+from djl_python.rolling_batch.vllm_rolling_batch_base import VllmRollingBatchBase, DTYPE_MAPPER
+from djl_python.properties_manager.vllm_rb_properties import VllmRbProperties
+
+
+class LmiDistRollingBatch(VllmRollingBatchBase):
+    """
+    LmiDistRollingBatch connects handler to LmiDist backend engine. It receives new
+    requests from the handler and sends them to the backend when space is available in the batch.
+    It also gets any new tokens from the backend and sends them back to the handler.
+    """
+
+    def __init__(self, model_id_or_path: str, properties: dict, **kwargs):
+        """
+        Initializes the LmiDistRollingBatch.
+
+        :param model_id_or_path (str): Currently unused since there is a copy inside properties
+        :param properties (dict): other properties of the model, such as decoder strategy
+        """
+
+        engine_config = VllmRbProperties(**properties)
+        super().__init__(engine_config)
+        self.request_cache = OrderedDict()
+        self.init_engine()
+
+    def init_engine(self):
+        """
+        Initializes vllm engine
+        """
+        args = EngineArgs(
+            model=self.engine_config.model_id_or_path,
+            tensor_parallel_size=self.engine_config.tensor_parallel_degree,
+            dtype=DTYPE_MAPPER[self.engine_config.dtype],
+            seed=0,
+            max_model_len=self.engine_config.max_model_len,
+            enforce_eager=self.engine_config.enforce_eager,
+            gpu_memory_utilization=self.engine_config.gpu_memory_utilization,
+            max_num_batched_tokens=self.engine_config.
+            max_rolling_batch_prefill_tokens,
+            trust_remote_code=self.engine_config.trust_remote_code,
+            load_format=self.engine_config.load_format,
+            quantization=self.engine_config.quantize,
+            revision=self.engine_config.revision)
+        self.engine = engine_from_args(args)
+
+    def add_request(self, request_id: str, prompt: str,
+                    sampling_params: SamplingParams):
+        """
+        Adds request to the engine
+        """
+        lmi_dist_request = Request(id=request_id,
+                                   prompt=prompt,
+                                   sampling_params=sampling_params)
+        self.engine.add_request(lmi_dist_request)
+
+    def translate_to_engine_params(self, parameters: dict):
+        """
+        Helper function to convert DJL Serving parameter names to parameter names
+        that lmidist_v2 recognizes.
+
+        :param parameters (dict): Parameters pertaining to a specific request
+
+        :return: The same parameters dict, but with VLLM style parameter names.
+        """
+        parameters.pop('seed', None)
+        parameters.pop('do_sample', None)
```

Comment:
As I said above, I used vllm config here. But it's a good point, I believe this is removed for `vllm` because it doesn't support `do_sample` parameter whereas `lmi-dist` should (for backwards compatibility) and we should set default sampling params

Reply:
Sure, we can punt on this, but I think this is another point to bring up about how consistent of an interface we want to provide across engine/backend, vs. how closely the interface should change to match each engine/backend.